src/test/scala/com/mongodb/spark/config/WriteConfigSpec.scala

```diff
@@ -52,17 +59,17 @@ class WriteConfigSpec extends FlatSpec with Matchers {
   it should "validate the values" in {
     an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().remove("mongodb.output.databaseName"))
     an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().remove("mongodb.output.collectionName"))
-    an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().set("mongodb.output.writeConcern", "allTheNodes"))
-    an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().set("mongodb.output.writeConcern", "{Fail}"))
+    an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().set("mongodb.output.writeConcern.w", "-1"))
+    an[IllegalArgumentException] should be thrownBy WriteConfig(sparkConf.clone().set("mongodb.output.writeConcern.wTimeoutMS", "-1"))
   }
 
   val sparkConf = new SparkConf()
     .set("mongodb.output.databaseName", "db")
     .set("mongodb.output.collectionName", "collection")
-    .set("mongodb.output.writeConcern", "majority")
 
   val writeConcerns = Table(
     "writeConcern",
```

Comment:
Should test a user-supplied w value, like:

  new WriteConcern("userDefinedValue")